File: app/src/main/java/com/datarockets/mnchkn/store/PlayerServiceImpl.java
package com.datarockets.mnchkn.store;

import android.content.Context;

import com.datarockets.mnchkn.MunchkinApplication;
import com.datarockets.mnchkn.models.Player;
import com.datarockets.mnchkn.utils.ColorUtil;
import com.datarockets.mnchkn.utils.LogUtil;

import java.util.ArrayList;

import javax.inject.Inject;

public class PlayerServiceImpl implements PlayerService {

    public static final String TAG = LogUtil.makeLogTag(PlayerServiceImpl.class);

    @Inject
    MunchkinDatabaseHelper mPlayerDatabase;

    public PlayerServiceImpl(Context context) {
        MunchkinApplication.get(context).getAppComponent().inject(this);
    }

    @Override
    public Player addPlayer(String name) {
        Player player = new Player();
        player.setName(name);
        player.setLevelScore(1);
        player.setStrengthScore(1);
        player.setColor(ColorUtil.generatePlayerAvatarColor());
        long id = mPlayerDatabase.addPlayer(player);
        player.setId(id);
        return player;
    }

    @Override
    public int deletePlayer(int position, long id) {
        mPlayerDatabase.deletePlayer(id);
        return position;
    }

    @Override
    public Player updatePlayer(Player player) {
        return mPlayerDatabase.updatePlayer(player);
    }

    @Override
    public void clearPlayersStats() {
        mPlayerDatabase.clearPlayersStats();
    }

    @Override
    public ArrayList<Player> getPlayersList() {
        return mPlayerDatabase.getPlayers();
    }

    @Override
    public ArrayList<Player> getPlayersList(int orderValue) {
        return mPlayerDatabase.getPlayers(orderValue);
    }

}

File: app/src/main/kotlin/com/datarockets/mnchkn/store/PlayerServiceImpl.kt
package com.datarockets.mnchkn.store

import android.content.Context
import com.datarockets.mnchkn.MunchkinApplication
import com.datarockets.mnchkn.models.Player
import com.datarockets.mnchkn.utils.ColorUtil
import com.datarockets.mnchkn.utils.LogUtil
import java.util.*
import javax.inject.Inject

class PlayerServiceImpl(context: Context) : PlayerService {

    @Inject
    lateinit var mPlayerDatabase: MunchkinDatabaseHelper

    init {
        MunchkinApplication.get(context).appComponent.inject(this)
    }

    override fun addPlayer(name: String): Player {
        val player = Player(id = 0, totalScore = 0, name = name, levelScore = 1, strengthScore = 1, color = ColorUtil.generatePlayerAvatarColor())
        val id = mPlayerDatabase.addPlayer(player)
        player.id = id
        return player
    }

    override fun deletePlayer(position: Int, id: Long): Int {
        mPlayerDatabase.deletePlayer(id)
        return position
    }

    override fun updatePlayer(player: Player): Player {
        return mPlayerDatabase.updatePlayer(player)
    }

    override fun clearPlayersStats() {
        mPlayerDatabase.clearPlayersStats()
    }

    override val playersList: ArrayList<Player>
        get() = mPlayerDatabase.players

    override fun getPlayersList(orderValue: Int): ArrayList<Player> {
        return mPlayerDatabase.getPlayers(orderValue)
    }

    companion object {

        val TAG = LogUtil.makeLogTag(PlayerServiceImpl::class)
    }

}
